hashexpr/src/lib.rs

```diff
@@ -149,20 +155,22 @@ impl Expr {
         ret.extend(vec![size_byte]);
         ret.extend(xs_count);
         ret.extend(xs.iter().fold(vec![], |mut acc, x| {
-          acc.extend(Expr::serialize(x));
+          acc.extend(Self::serialize(x));
           acc
         }));
         ret
       }
     }
   }
 
+  #[must_use]
   pub fn link(&self) -> Link { Link::make(&self.serialize()) }
 
-  pub fn hash(&self) -> Expr { link!(self.position(), self.link()) }
+  #[must_use]
+  pub fn hash(&self) -> Self { link!(self.position(), self.link()) }
 
-  pub fn deserialize(i: &[u8]) -> IResult<&[u8], Expr, DeserialError<&[u8]>> {
-    let (i, size) = take(1 as usize)(i)?;
+  pub fn deserialize(i: &[u8]) -> IResult<&[u8], Self, DeserialError<&[u8]>> {
+    let (i, size) = take(1_usize)(i)?;
```

Comment:
Sure. It is exactly the same as `1usize` (in Rust, you can insert an arbitrary number of underscores in number literals). IMO it is more clear than `as` because instead of creating an `i32` (the default type of number literals in Rust) and then casting it to a `usize`, we are creating a `usize` out of the box. I don't think they would compile to the same thing, no.
https://rust-lang.github.io/rust-clippy/master/index.html#unnecessary_cast